webapp/tests/Unit/Controller/API/ProblemControllerTest.php

```diff
@@ -13,39 +13,60 @@ class ProblemControllerTest extends BaseTest
     protected ?string $apiEndpoint = 'problems';
 
     protected array $expectedObjects = [
-      3 => [
-        "ordinal" => 0,
-        "id" => "3",
-        "short_name" => "boolfind",
-        "label" => "boolfind",
-        "time_limit" => 5,
-        "externalid" => "boolfind",
-        "name" => "Boolean switch search",
-        "rgb" => "#008000",
-        "color" => "green",
-      ],
-      2 => [
-        "ordinal" => 1,
-        "id" => "2",
-        "short_name" => "fltcmp",
-        "label" => "fltcmp",
-        "time_limit" => 5,
-        "externalid" => "fltcmp",
-        "name" => "Float special compare test",
-        "rgb" => "#CD5C5C",
-        "color" => "indianred",
-      ],
-      1 => [
-        "ordinal" => 2,
-        "id" => "1",
-        "short_name" => "hello",
-        "label" => "hello",
-        "time_limit" => 5,
-        "externalid" => "hello",
-        "name" => "Hello World",
-        "rgb" => "#87CEEB",
-        "color" => "skyblue",
-      ],
+        3 => [
+            "ordinal"    => 0,
+            "id"         => "3",
+            "short_name" => "boolfind",
+            "label"      => "boolfind",
+            "time_limit" => 5,
+            "externalid" => "boolfind",
+            "name"       => "Boolean switch search",
+            "rgb"        => "#008000",
+            "color"      => "green",
+            "statement" => [
+                [
+                    'href'     => 'contests/2/problems/3/statement',
+                    'mime'     => 'application/pdf',
+                    'filename' => 'statement.pdf',
+                ],
+            ],
+        ],
+        2 => [
+            "ordinal"    => 1,
+            "id"         => "2",
+            "short_name" => "fltcmp",
+            "label"      => "fltcmp",
+            "time_limit" => 5,
+            "externalid" => "fltcmp",
+            "name"       => "Float special compare test",
+            "rgb"        => "#CD5C5C",
+            "color"      => "indianred",
+            "statement" => [
+                [
+                    'href'     => 'contests/2/problems/2/statement',
+                    'mime'     => 'application/pdf',
+                    'filename' => 'statement.pdf',
+                ],
+            ],
+        ],
+        1 => [
+            "ordinal"    => 2,
+            "id"         => "1",
+            "short_name" => "hello",
+            "label"      => "hello",
+            "time_limit" => 5,
+            "externalid" => "hello",
+            "name"       => "Hello World",
+            "rgb"        => "#87CEEB",
+            "color"      => "skyblue",
+            "statement" => [
+                [
+                    'href'     => 'contests/2/problems/1/statement',
+                    'mime'     => 'application/pdf',
+                    'filename' => 'statement.pdf',
+                ],
+            ],
+        ],
```

Comment:
We should consider adding a test to show that you cannot ask for the statement of a problem not in this contest.

Reply:
I think I already replied in Slack, but this specific test is not for the statements, but more general: you should not be able to access a problem from another contest in some contest. I did add some other tests though.